pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -0,0 +1,296 @@
+		operatorConfigName: operatorConfigName,
+
+		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "PruneController"),
+		prunerPodImageFn: getPrunerPodImageFromEnv,
+	}
+
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *PruneController) syncRevisionLimits() error {
+	config, err := c.dynamicClient.Resource(c.operatorConfigGVR).Get(c.operatorConfigName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	revisionLimits, ok, err := unstructured.NestedMap(config.Object, "spec", "operatorRevisionLimit")
+	if err != nil {
+		return err
+	}
+	if ok {
+		if val, set := revisionLimits["failed"]; set {
+			c.failedRevisionLimit = val.(int)
+		}
+		if val, set := revisionLimits["succeeded"]; set {
+			c.succeededRevisionLimit = val.(int)
+		}
+	}
+	return nil
+}
+
+func (c *PruneController) pruneRevisionHistory(operatorStatus *operatorv1.StaticPodOperatorStatus, resourceDir string) error {
+	var succeededRevisionIDs, failedRevisionIDs []int
+
+	configMaps, err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, configMap := range configMaps.Items {
+		if strings.HasPrefix(configMap.Name, statusConfigMapName) {
+			if revision, ok := configMap.Data["revision"]; ok {
+				revisionID, err := strconv.Atoi(revision)
+				if err != nil {
+					return err
+				}
+				switch configMap.Data["phase"] {
+				case string(corev1.PodSucceeded):
+					succeededRevisionIDs = append(succeededRevisionIDs, revisionID)
+				case string(corev1.PodFailed):
+					failedRevisionIDs = append(failedRevisionIDs, revisionID)
+				}
+			}
+		}
+	}
+
+	sort.Ints(succeededRevisionIDs)
+	sort.Ints(failedRevisionIDs)
+
+	// Get list of protected IDs
+	protectedSucceededRevisionIDs := succeededRevisionIDs[revisionKeyToStart(len(succeededRevisionIDs), c.succeededRevisionLimit):]
+	protectedFailedRevisionIDs := failedRevisionIDs[revisionKeyToStart(len(failedRevisionIDs), c.failedRevisionLimit):]
+
+	excludedIDs := make([]int, 0, len(protectedSucceededRevisionIDs)+len(protectedFailedRevisionIDs))
+	excludedIDs = append(excludedIDs, protectedSucceededRevisionIDs...)
+	excludedIDs = append(excludedIDs, protectedFailedRevisionIDs...)
+	sort.Ints(excludedIDs)
+
+	// Run pruning pod on each node and pin it to that node
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		err = c.ensurePrunePod(prunerPodName, nodeStatus.NodeName, excludedIDs[len(excludedIDs)-1], excludedIDs, nodeStatus.TargetRevision)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func revisionKeyToStart(length, limit int) int {
+	if length > limit {
+		return length - limit
+	}
+	return 0
+}
+
+func (c *PruneController) ensurePrunePod(podName, nodeName string, maxEligibleRevision int, protectedRevisions []int, revision int32) error {
+	required := resourceread.ReadPodV1OrDie([]byte(prunerPod))
```

Comment:
i think we will prefer to use bindata for manifests instead of embedding the manifests in code via const.

see https://github.com/openshift/library-go/pull/145 for example

/cc @sttts